src/redis-check-aof.c

```diff
@@ -51,45 +67,138 @@ int consumeNewline(char *buf) {
     return 1;
 }
 
-int readAnnotations(FILE *fp) {
-    char buf[AOF_ANNOTATION_LINE_MAX_LEN];
+#define AOF_MANIFEST_KEY_FILE_NAME   "file"
+#define AOF_MANIFEST_KEY_FILE_SEQ    "seq"
+#define AOF_MANIFEST_KEY_FILE_TYPE   "type"
+#define AOF_MANIFEST_MAX_LINE        1024
+static aofManifest *loadManifestFromFile(char *am_filepath) {
+    const char *err = NULL;
+    long long maxseq = 0;
+
+    aofManifest *am = aofManifestCreate();
+
+    if (!fileExist(am_filepath)) {
+        printf("The AOF manifest file %s doesn't exist", am_filepath);
+        exit(1);
+    }
+
+    FILE *fp = fopen(am_filepath, "r");
+    if (fp == NULL) {
+        printf("Fatal error: can't open the AOF manifest "
+            "file %s for reading: %s", am_filepath, strerror(errno));
+        exit(1);
+    }
+
+    char buf[AOF_MANIFEST_MAX_LINE+1];
+    sds *argv = NULL;
+    int argc;
+    aofInfo *ai = NULL;
+
+    sds line = NULL;
+    int linenum = 0;
+
     while (1) {
-        epos = ftello(fp);
-        if (fgets(buf, sizeof(buf), fp) == NULL) {
-            return 0;
-        }
-        if (buf[0] == '#') {
-            if (to_timestamp && strncmp(buf, "#TS:", 4) == 0) {
-                time_t ts = strtol(buf+4, NULL, 10);
-                if (ts <= to_timestamp) continue;
-                if (epos == 0) {
-                    printf("AOF has nothing before timestamp %ld, "
-                           "aborting...\n", to_timestamp);
-                    fclose(fp);
-                    exit(1);
-                }
-                /* Truncate remaining AOF if exceeding 'to_timestamp' */
-                if (ftruncate(fileno(fp), epos) == -1) {
-                    printf("Failed to truncate AOF to timestamp %ld\n",
-                            to_timestamp);
-                    exit(1);
+        if (fgets(buf, AOF_MANIFEST_MAX_LINE+1, fp) == NULL) {
+            if (feof(fp)) {
+                if (linenum == 0) {
+                    err = "Found an empty AOF manifest";
+                    goto loaderr;
                 } else {
-                    printf("Successfully truncated AOF to timestamp %ld\n",
-                            to_timestamp);
-                    fclose(fp);
-                    exit(0);
+                    break;
                 }
+            } else {
+                err = "Read AOF manifest failed";
+                goto loaderr;
             }
-            continue;
-        } else {
-            if (fseek(fp, -(ftello(fp)-epos), SEEK_CUR) == -1) {
-                ERROR("Fseek error: %s", strerror(errno));
-                return 0;
+        }
+
+        linenum++;
+
+        /* Skip comments lines */
+        if (buf[0] == '#') continue;
```

Comment:
We only support like this:

```
# Some annotations
file appendonly.aof.1.base.rdb seq 1 type b
# Some annotations
file appendonly.aof.2.incr.aof seq 2 type i
# Some annotations
# Some annotations
```

Reply:
First, in the context of manifest files (not AOF files) these are more "comments" and not "annotations", right? Also I guess for now this is OK, perhaps we'd like to improve this in the future, but I guess as long an we'll be backwards compatible it'll be ok. Again, @oranagra WDYT?

Reply:
Yes, I think we need to leave room for manifest to expand, I think allowing `# `to appear is a method similar to using annotation in AOF, similar to using annotation in AOF.

BTW It would be fine if we used it as a comment, like explaining the meaning of the file below, although we don't currently do that.